### PR TITLE
Address machineType not being returned in GCP metadata

### PR DESCRIPTION
**Pull request by picandocodigo**
Related: https://github.com/elastic/apm/issues/847

### CODE REVIEW

File: lib/elastic_apm/metadata/cloud_info.rb [hunk 1]
```diff
@@ -102,7 +102,7 @@ def fetch_gcp
         self.project_id = metadata["project"]["projectId"]
         self.availability_zone = zone
         self.region = zone.split("-")[0..-2].join("-")
-        self.machine_type = metadata["instance"]["machineType"].split("/")[-1]
+        self.machine_type = metadata["instance"]["machineType"]&.split("/")&.[](-1)
```
Comment (estolfo) [inline lib/elastic_apm/metadata/cloud_info.rb:105]: Should we use Hash#dig instead in case `"instance"` doesn't exist for some reason?

Comment (picandocodigo) [reply lib/elastic_apm/metadata/cloud_info.rb:105]: Some previous values depend on `instance`, should I rewrite it and update it to use `fetch` in the whole function or just for `machineType`? It does feel more readable than the safe navigation operator, so I'll update at least this line to `fetch` anyway.

Comment (estolfo) [reply lib/elastic_apm/metadata/cloud_info.rb:105]: Ah right, we use `instance` above already. It's up to you, whatever you think is most readable and consistent.

Comment (picandocodigo) [reply lib/elastic_apm/metadata/cloud_info.rb]: I updated it to return early if `instance` doesn't exist, and I changed the code to look just like the code that sets the zone so these two lines are consistent:
```ruby
zone = metadata["instance"]["zone"]&.split("/")&.at(-1)

self.machine_type = metadata["instance"]["machineType"]&.split("/")&.at(-1)
```

**Review by estolfo — APPROVED** [review]

File: lib/elastic_apm/metadata/cloud_info.rb (hunk 1 above)
Comment (estolfo) [reply lib/elastic_apm/metadata/cloud_info.rb]: perfect, thank you!

**Pull request merged**